apps/src/templates/sectionProgressV2/ExpandedProgressDataColumn.jsx

```diff
@@ -39,64 +43,76 @@ function ExpandedProgressDataColumn({
   };
 
   const getSingleLevelColumn = React.useCallback(
-    (level, propOverrides = {}) => {
+    (level, studentId, propOverrides = {}) => {
       return (
-        <div
-          className={styles.expandedLevelColumn}
-          key={lesson.id + '.' + level.id}
-        >
-          {sortedStudents.map(student => (
-            <LevelDataCell
-              studentId={student.id}
-              level={level}
-              studentLevelProgress={
-                levelProgressByStudent[student.id][level.id]
-              }
-              key={student.id + '.' + lesson.id + '.' + level.id}
-              {...propOverrides}
-            />
-          ))}
-        </div>
+        <LevelDataCell
+          studentId={studentId}
+          level={level}
+          studentLevelProgress={levelProgressByStudent[studentId][level.id]}
+          key={studentId + '.' + lesson.id + '.' + level.id}
+          lessonId={lesson.id}
+          {...propOverrides}
+        />
       );
     },
-    [levelProgressByStudent, sortedStudents, lesson]
+    [levelProgressByStudent, lesson]
   );
 
   const getExpandedChoiceLevel = React.useCallback(
-    level => [
-      getSingleLevelColumn(level, {expandedChoiceLevel: true}),
-      ...level.sublevels.map(sublevel => getSingleLevelColumn(sublevel)),
+    (level, studentId) => [
+      getSingleLevelColumn(level, studentId, {expandedChoiceLevel: true}),
+      ...level.sublevels.map(sublevel =>
+        getSingleLevelColumn(sublevel, studentId, {parentLevelId: level.id})
+      ),
     ],
     [getSingleLevelColumn]
   );
 
   const progress = React.useMemo(
     () => (
-      <div className={styles.expandedTable}>
-        {lesson.levels.flatMap(level => {
-          if (
-            level.sublevels?.length > 0 &&
-            expandedChoiceLevels.includes(level.id)
-          ) {
-            return getExpandedChoiceLevel(level);
-          }
-          return [getSingleLevelColumn(level)];
-        })}
-      </div>
+      <tbody className={styles.expandedTable}>
+        {sortedStudents.map(student => (
+          <>
+            <th hidden={true} id={getStudentRowHeaderId(student.id)}>
+              {getFullName(student)}
```

Comment:
I was hoping to see some sort of name retrieval after reading about the id- nice!